rfc/rfc021-vp_token-grant-type.md

```diff
@@ -0,0 +1,162 @@
+2. The authorization server MUST validate the VP according to section 6.1 of the Presentation Exchange [PE] specification.
+3. The Presentation Submission MUST be an answer to the Presentation Definition that corresponds with the requested scope. 
+   How an authorization server determines the Presentation Definition based on scope is out of scope of this specification.
+   However, a request to the Presentation Definition endpoint MUST result to the same Presentation Definition as the Presentation Definition that is used to validate the VP.
+4. The Verifiable Presentation MUST be not be valid for more than 15 minutes.
+5. A clock skew of no more than 5 seconds MAY be applied when validating the Verifiable Presentation.
+
+### 4.2 JWT requirements
+
+1. The assertion MUST be a valid JWT according to §6.3.1 of the Verifiable Credentials Data Model 1.1 [VC-DATA-MODEL].
+2. The `vp` field of the JWT MUST be valid according to §4.10 the Verifiable Credentials Data Model 1.1 [VC-DATA-MODEL].
+3. The `nonce` of the JWT MUST be a string that is unique for each request.
+4. The `kid` field MUST be a DID URL.
+5. The `iss` field MUST be a Decentralized Identifier [DID] and MUST correspond to the `kid`.
+6. The `sub` field MUST match the `credentialSubject.ID` field from all the Verifiable Credentials that are used to request the access token.
+7. The `aud` field MUST be a DID under control of the Authorization Server.
+
+### 4.3 JSON requirements
+
+1. The assertion MUST be a valid JSON object according to §4.10 the Verifiable Credentials Data Model 1.1 [VC-DATA-MODEL].
+2. The proof of the VP MUST be a valid [JSONWebSignature2020] object.
+3. The `challenge` field of the JSON object MUST be a string that is unique for each request.
+4. The `domain` field of the JSON object MUST be a DID under control of the Authorization Server.
+5. The `issuer` field of the JSON object MUST be a Decentralized Identifier [DID].
+6. The `verificationMethod` field of the Proof MUST be a DID URL and correspond to the `issuer` field.
+7. The `holder` field if present MUST match the `credentialSubject.ID` field from all the Verifiable Credentials that are used to request the access token.
+8. If the `holder` field is not present then the `issuer` field MUST match the `credentialSubject.ID` field from all the Verifiable Credentials that are used to request the access token.
+
+## 5. Presentation Definition endpoint
+
+In order for a client to know which Presentation Definition [PE] to use, the authorization server MUST provide a Presentation Definition endpoint.
+The Presentation Definition endpoint MUST be registered as a `presentation_definition_endpoint` in the authorization server metadata [RFC8414].
+The Presentation Definition endpoint MUST return a Presentation Definition that corresponds with the requested scope.
+
+## 6. Access Token Introspection
+
+The authorization server MUST support the access token introspection endpoint as defined in OAuth 2.0 [RFC7662].
```

Comment:
Not sure whether this should be in scope of this RFC, since it's not relevant for the client. I would specify it as if the AS were to support Token Introspection, the fields SHOULD map as follows